### ISSUE TITLE
Unable to reproduce reported accuracy

### ISSUE THREAD

**Issue by kimdanni** [chaudatascience/cha_mae_vit#3]
We conducted experiments using the officially released code, following the exact experimental settings described in the paper, with the goal of achieving an accurate reproduction of the reported results. Using the same training script and environment, the reproduced test accuracy reached 83.26, which falls short of the reported 90.73.

Could you please let us know if you are aware of any possible reasons for this performance gap, or share any additional implementation details or settings that might help us reproduce the reported results more accurately?

We used the script below and have attached the training logs for your reference.

`accelerate launch --main_process_port=9999 --num_processes=2 main.py ++model.training_sample=DCP ++model.init_values=1 ++model.patch_size=16 ++train.max_grad_norm=1 ++model.attention_cls=v2 ++train.num_workers=3 ++train.learning_rate=0.0004 ++logging.run_name=JTestCodeChAMAEViT ++train.batch_size=32 ++eval.eval_batch_size=300 ++data.chammiv1.normalization.use_mean_std=True ++model.mae_lambda=0.99 ++model.mae_loss_norm=norm_pix_loss ++model.model_size=small ++eval.eval_every_epochs=5 ++model.decoder.dcp_p_patch=0.5 ++model.decoder.dcp_p_channel=0.5 ++model.decoder.dcp_mask_ratio=0.25 ++model.num_memory_tokens=4 ++model.proxy_loss_lambda=0 ++model.use_self_image_norm=False ++eval.eval_before_training=False ++train.checkpointing_limit=1 ++model.use_proxy_head=False ++model.encoder_pooling=cls_attn_pooling_sigmoid ++model.diverse_patch_token_weight=0.000001 ++model.diverse_channel_token_weight=0.001 ++model.diverse_patch_token_gamma_s=1 ++model.diverse_patch_token_gamma_d=4 ++data.training_dataset=jumpcp ++train.weight_decay=0.04 ++model.depth=11 ++model.cross_entropy_lambda=0.01 ++train.lr_warmup_epochs=10 ++train.seed=33677 `

[training.log](https://github.com/user-attachments/files/23493518/training.log)

**Comment by chaudatascience** [comment]
Hi,

Thanks for letting us know!
I'll check on it soon and will get back to you.

Best,
Chau

**Comment by chaudatascience** [comment]
Hi,

In the attached log, the accuracy after epoch 10 is very low (`acc_all 24.33116912841797`). It should be around 33% instead.

I noticed some differences in the config. I have updated the default configuration (the YAML file). Please pull the new code (with the updated config) and try again with the same command.

I am also re-running the code to make sure the results are reproducible. The job is still running, but the results look good (at epoch 10, accuracy is 33.006%). I'll upload the log of the job once it's done, which should be in ~30 hours from now.

Just let me know if the issue persists.

Best,
Chau

**Comment by kimdanni** [comment]
Thank you sincerely for always being so kind and helpful despite your busy schedule.
I believe that your research is making a significant contribution to the advancement of the multichannel research field.

**Comment by chaudatascience** [comment]
Thanks for the kind words!

I updated the code slightly by changing a temperature setting to match the previous version before cleaning up. Then, I reran the code and got results similar to those reported in the paper.

I uploaded the training logs in the [`training_logs`](https://github.com/chaudatascience/cha_mae_vit/tree/main/training_logs) folder for your reference. 

Please use the newest version of the code with the command in [`train_scripts.sh`](https://github.com/chaudatascience/cha_mae_vit/blob/main/train_scripts.sh) and let me know if you can reproduce the results.

Have a nice day!
Best,
Chau

**Comment by chaudatascience** [comment]
Hi, 

I hope it resolved the issue.
I'll close this thread here, but feel free to make a new one if you have any other questions.

Best,
Chau
